### DISCUSSION TITLE
Locations of multiple schema files

### DISCUSSION THREAD

**Discussion by geekox86** [dotansimha/graphql-code-generator#5055] [Q&A] [answered]
Greetings. Thank you for the huge efforts here.

I am learning to write custom plugins.
My plugin should generate a separate output file per input GraphQL schema file. But I have no clue where to get the locations of these input schema files.

Pleas, help me. Thanks.

**Comment by dotansimha** [comment, marked as answer]
Hi @geekox86 ! 
We recently found that we need something similar for [this PR](https://github.com/dotansimha/graphql-code-generator/pull/3923). We added `includeSources: boolean` flag [to `graphql-tools` loaders](https://github.com/ardatan/graphql-tools/blob/df52bf8cfab7ec796546929e7d50022652b222ef/packages/load/src/schema.ts#L10-L15), and it should already be available for you to use. 

I *think* that doing something like that should work:

```yml
schema:
  - "./src/**/*.graphql": 
     includeSources: true
```

And then you can use the `GraphQLSchema` and access `schema.extensions.sources` to get the sources that created the schema.

Can you please try that and let me know how it works? 

fyi @ardatan

**Reply by geekox86** [reply]
Thank you @dotansimha. Is there a way to transform individual GraphQL schema files to some other format without actually having to specify them one by one in the codegen config file as outputs? The problem is that my custom plugin can only return one string.

**Reply by dotansimha** [reply]
Can you please elaborate? what is the API/data you wish to have?
